src/components/TextInput/__tests__/TextInput.spec.js

```diff
@@ -120,6 +120,33 @@ describe('Text input', () => {
     expect(slot).toBeInTheDocument();
   });
 
+  it('renders the helper text correctly', () => {
+    const props = {
+      ...initialProps,
+      helperText: 'Helper text'
+    };
+    const { getByText } = render(TextInput, {
+      props
+    });
+
+    const helperText = getByText('Helper text');
+    expect(helperText).toBeInTheDocument();
+  });
+
+  it('clears the input when the clear button is clicked', async () => {
+    const props = initialProps;
+    const { getByRole, getByLabelText } = render(TextInput, { props });
+
+    const textInput = getByLabelText(props.label);
+    const updatedValue = 'hello!';
+    await fireEvent.update(textInput, updatedValue);
```

Comment:
I'm not suggesting you change this now but more just sharing for future reference.  I think the equivalent to this in `userEvent` is `type`: https://testing-library.com/docs/user-event/utility#type